api/lib/source/cosmosV0-source.js

```diff
@@ -224,14 +224,106 @@ class CosmosV0API extends RESTDataSource {
       validator.signing_info = signingInfos[consensusAddress]
     })
 
-    return validators.map((validator) =>
+    const validatorsWithoutProfiles = validators.map((validator) =>
       this.reducers.validatorReducer(
         this.network.id,
         signedBlocksWindow,
         validator,
         annualProvision
       )
     )
+    if (profile) {
+      return { validators, validatorsWithoutProfiles }
+    } else {
+      return validatorsWithoutProfiles
+    }
+  }
+
+  async getProfilesForValidators(
+    validators,
+    validatorsWithoutProfiles,
+    fiatCurrency = 'USD'
+  ) {
+    validatorsWithoutProfiles = this.getRanksForValidators(
+      validatorsWithoutProfiles
+    )
+    return await Promise.all(
+      validatorsWithoutProfiles.map(async (enrichedValidator) => {
+        const [validatorProfile, allValidatorDelegations] = await Promise.all([
+          this.db.getValidatorProfile(enrichedValidator.operatorAddress),
```

Comment:
why do we need to query the validator profile here again? we already enrich validator profiles when we put them into the store, right?

Reply:
no, we need to get all the social networks & validator metadata from DB. Look at the validatorprofiles table in staging DB